From 55d16cdbce74691e1aebb958b9d3067d845d6436 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 17 Apr 2017 13:07:48 -0400
Subject: manager: fix crash if session fails to start

commit 5c9e120594839b0597bc7bb8d06be7ba1076c0d8 attempts to handle
a session failing to start, but messages up the signal prototype
leading to crash.

This commit fixes that.

https://bugzilla.gnome.org/show_bug.cgi?id=781413
---
 daemon/gdm-manager.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/daemon/gdm-manager.c b/daemon/gdm-manager.c
index b909976..b9b230a 100644
--- a/daemon/gdm-manager.c
+++ b/daemon/gdm-manager.c
@@ -1976,9 +1976,10 @@ remove_user_session (GdmManager *manager,
 static void
 on_session_start_failed (GdmSession *session,
                          const char *service_name,
+                         const char *message,
                          GdmManager *manager)
 {
-        g_debug ("GdmManager: session failed to start");
+        g_debug ("GdmManager: session failed to start: %s", message);
         remove_user_session (manager, session);
 }
 

